src/System.Private.CoreLib/shared/System/Diagnostics/DebugProvider.Unix.cs

```diff
@@ -10,8 +10,14 @@ public partial class DebugProvider
     {
         private static readonly bool s_shouldWriteToStdErr = Environment.GetEnvironmentVariable("COMPlus_DebugWriteToStdErr") == "1";
 
-        public virtual void ShowDialog(string stackTrace, string message, string detailMessage, string errorSource)
+        public static void ShowDialog(string stackTrace, string message, string detailMessage, string errorSource)
         {
+            if (s_ShowDialog != null)
+            {
+                s_ShowDialog(stackTrace, message, detailMessage, errorSource); 
```

Comment:
If this can be set to null, there's a potential race condition here, and s_showDialog should be stored into a local first and then that local checked and used.

Reply:
`s_ShowDialog` is a hook for our existing internal tests only.